validator/client/validator.go

```diff
@@ -116,63 +123,73 @@ func (v *validator) NextSlot() <-chan uint64 {
 // list of upcoming assignments needs to be updated. For example, at the
 // beginning of a new epoch.
 func (v *validator) UpdateAssignments(ctx context.Context, slot uint64) error {
-	if slot%params.BeaconConfig().SlotsPerEpoch != 0 && v.assignment != nil {
+	if slot%params.BeaconConfig().SlotsPerEpoch != 0 && v.assignments != nil {
 		// Do nothing if not epoch start AND assignments already exist.
 		return nil
 	}
 
 	ctx, span := trace.StartSpan(ctx, "validator.UpdateAssignments")
 	defer span.End()
-
+	pks := [][]byte{}
+	for i := range v.keys {
+		pks = append(pks, v.keys[i].PublicKey.Marshal())
+	}
 	req := &pb.CommitteeAssignmentsRequest{
 		EpochStart: slot,
-		PublicKey:  [][]byte{v.key.PublicKey.Marshal()},
+		PublicKey:  pks,
 	}
 
 	resp, err := v.validatorClient.CommitteeAssignment(ctx, req)
 	if err != nil {
-		v.assignment = nil // Clear assignments so we know to retry the request.
+		v.assignments = nil // Clear assignments so we know to retry the request.
 		return err
 	}
 
-	v.assignment = resp
+	v.assignments = resp
+	for i := range v.assignments.Assignment {
```

Comment:
why not use this way ?
```go 
for i,assignment := range v.assignments.Assignment {

```
would not need to be so verbose below

Reply:
done , thanks